File: Data Structures & Algorithms/house-robber/submission-1.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        memo = {}
        def solve(index) : 
            if index <= -1 :
                return 0
            
            if index in memo : 
                return memo[index]
                
            memo[index] = max(nums[index] + solve(index - 2), solve(index - 1))
            return memo[index]

        
        return solve(len(nums) - 1)

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;

        unordered_map<long long, int> cons;
        long long min_el = 1e9;
        long long max_el = -1e9;
        
        for (int i = 0; i < nums.size(); i++) {
           cons[nums[i]] = 1;
           min_el = min(min_el, (long long)nums[i]); 
           max_el = max(max_el, (long long)nums[i]);
        }

        int max_seq = 0;
        int sub_seq = 0;
        
        for (int i = min_el; i < max_el + 1; i++) { 
            if (cons.contains(i)) {
                sub_seq++;
            } else {
                max_seq = max(max_seq, sub_seq);
                sub_seq = 0;
            }
        }

        return max(max_seq, sub_seq); 
    }
};